src/python/pants/backend/docker/goals/package_image.py

```diff
@@ -343,6 +347,16 @@ def get_build_options(
                 DockerBuildOptionFlagFieldMixin,
             ),
         ):
+            if ignore_platforms and field_type == DockerImageBuildPlatformOptionField:
```

Comment:
We need to apply slightly different options to `package` and `publish`. Theres probably a nicer way I could clean this up